x-pack/plugins/observability_solution/inventory/public/components/entities_grid/index.tsx

```diff
@@ -137,15 +64,36 @@ export function EntitiesGrid({
     [onChangeSort]
   );
 
+  const showAlertsColumn = useMemo(
+    () => entities?.some((entity) => entity?.alertsCount && entity?.alertsCount > 0),
+    [entities]
+  );
+
+  const visibleColumns = useMemo(
+    () => getColumns({ showAlertsColumn }).map(({ id }) => id),
+    [showAlertsColumn]
+  );
+
+  const columnVisibility = useMemo(
+    () => ({
+      visibleColumns,
+      setVisibleColumns: () => {},
+    }),
+    [visibleColumns]
+  );
```

Comment:
can you merge these?

Reply:
https://github.com/elastic/kibana/pull/195250/commits/9197530b2ea7b3ce3aea72d572cb07e98dab22fe#diff-e2d526c922770507841d5cd7b257fd93b28321aae4a40ca436b38d234804334dR74